.ci/release/generate_build_matrix.py

```diff
@@ -0,0 +1,459 @@
+        accelerator_type (str): Type of accelerator (cpu, cuda, or rocm)
+        builds_on (str): Github Actions runner for building
+        tests_on (str): Github Actions runner for testing
+        arch_list (list[str]): List of specific architectures to build for
+        extra_install_requirements (list[str]): Additional packages needed for the build
+    """
+
+    operating_systems: list[OperatingSystem]
+    cpu_arch: CpuArch
+    accelerator_version: str
+    accelerator_type: str
+    builds_on: str = field(default="linux.12xlarge.memory.ephemeral")
+    tests_on: str = field(default="linux.4xlarge")
+    arch_list: list[str] = field(default_factory=list)
+    extra_install_requirements: list[str] = field(default_factory=list)
+
+    def container_image(self) -> str:
+        """
+        Returns the container image for the build.
+
+        Format: pytorch/wheel-build:{cpu_arch}-{accelerator_type}{accelerator_version}-{DEFAULT_TAG}
+        Produces images that look like:
+        - pytorch/wheel-build:x86_64-cpu-e8d4567
+        - pytorch/wheel-build:x86_64-cuda12.8-e8d4567
+        - pytorch/wheel-build:aarch64-cuda12.8-e8d4567
+        - pytorch/wheel-build:x86_64-rocm6.2.4-e8d4567
+        """
+        tag = _generate_docker_tag()
+        return f"pytorch/wheel-build:{str(self.cpu_arch)}-{self.accelerator_type}{self.accelerator_version}-{tag}"
+
+
+@dataclass
+class CpuBuild(BinaryBuild):
+    accelerator_type: str = "cpu"
+    accelerator_version: str = ""
+
+
+@dataclass
+class CudaBuild(BinaryBuild):
+    cudnn_version: str = "9"
+    accelerator_type: str = "cuda"
+    tests_on: str = "linux.4xlarge.nvidia.gpu"
+
+
+@dataclass
+class StableCudaBuild(CudaBuild):
+    is_stable: bool = True
+
+
+@dataclass
+class RocmBuild(BinaryBuild):
+    accelerator_type: str = "rocm"
+    tests_on: str = "linux.rocm.gpu"
+
+
+@dataclass
+class XpuBuild(BinaryBuild):
+    accelerator_type: str = "xpu"
+    accelerator_version: str = ""
+    builds_on: str = "linux.s390x"
+    tests_on: str = "linux.s390x"
+
+
+MANYWHEEL_BUILDS: list[BinaryBuild] = [
+    # CPU x86_64
+    CpuBuild(
+        operating_systems=[OperatingSystem.LINUX, OperatingSystem.WINDOWS],
+        cpu_arch=CpuArch.X86_64,
+    ),
+    # CPU ARM64
+    CpuBuild(
+        cpu_arch=CpuArch.AARCH64,
+        operating_systems=[
+            OperatingSystem.LINUX,
+            OperatingSystem.WINDOWS,
+            OperatingSystem.MACOS,
+        ],
+        builds_on="linux.arm64.m7g.4xlarge.ephemeral",
+        tests_on="linux.arm64.2xlarge",
+    ),
+    # CPU S390X
+    CpuBuild(
+        cpu_arch=CpuArch.S390X,
+        operating_systems=[OperatingSystem.LINUX],
+    ),
+    # NOTE: Also update the CUDA sources in tools/nightly.py when changing this list
+    # CUDA x86_64
+    CudaBuild(
+        accelerator_version="11.8",
+        cpu_arch=CpuArch.X86_64,
+        operating_systems=[OperatingSystem.LINUX, OperatingSystem.WINDOWS],
```

Comment:
I believe this section need to be included for all builds including mac builds. Otherwise poetry may break

Reply:
This should be updated now, basically I now add the x86 CUDA 12.6 extra install requirements for all builds aside from the ones that have their own extra install requirements

Reply:
After talking about this for a while internally we decided that builds of like accelerator type + version will have the same requirements and that for builds published to PyPI we will work later on a script to inject the dependencies of the latest stable CUDA version (the one we publish to PyPI) to the rest of the builds on PyPI